core/src/main/java/org/mule/processor/EndpointTransactionalInterceptingMessageProcessor.java

```diff
@@ -36,7 +38,7 @@ public MuleEvent process(final MuleEvent event) throws MuleException
         {
             return event;
         }
-        else
+        else if(TransactionCoordination.getInstance().getTransaction() != null)
```

Comment:
This chain that includes processor is constructed once for the endpoint but if a composite-source is being used, transactions may or may not be in used for a specific request, that is why this is needed here.